sormas-backend/src/main/java/de/symeda/sormas/backend/task/TaskFacadeEjb.java

```diff
@@ -519,52 +479,45 @@ public long count(TaskCriteria taskCriteria) {
 
 			Pseudonymizer pseudonymizer = Pseudonymizer.getDefault(userService::hasRight, I18nProperties.getCaption(Captions.inaccessibleValue));
 			Pseudonymizer emptyValuePseudonymizer = Pseudonymizer.getDefault(userService::hasRight);
-			pseudonymizer.pseudonymizeDtoCollection(
-				TaskIndexDto.class,
-				tasks,
-				t -> taskJurisdictionChecker.isInJurisdictionOrOwned(t.getJurisdiction()),
-				(t, ignored) -> {
-					if (t.getCaze() != null) {
-						emptyValuePseudonymizer.pseudonymizeDto(
-							CaseReferenceDto.class,
-							t.getCaze(),
-							caseJurisdictionChecker.isInJurisdictionOrOwned(t.getJurisdiction().getCaseJurisdiction()),
-							null);
-					}
-
-					if (t.getContact() != null) {
-						pseudonymizeContactReference(emptyValuePseudonymizer, t.getContact(), t.getJurisdiction().getContactJurisdiction());
-					}
-
-					if (t.getEvent() != null) {
-						emptyValuePseudonymizer.pseudonymizeDto(
-							EventReferenceDto.class,
-							t.getEvent(),
-							eventJurisdictionChecker.isInJurisdictionOrOwned(t.getJurisdiction().getEventJurisdiction()),
-							null);
-					}
-				},
-				true);
+			pseudonymizer.pseudonymizeDtoCollection(TaskIndexDto.class, tasks, t -> t.getInJurisdiction(), (t, ignored) -> {
+				if (t.getCaze() != null) {
+					emptyValuePseudonymizer.pseudonymizeDto(
+						CaseReferenceDto.class,
+						t.getCaze(),
+						caseService.inJurisdictionOrOwned(caseService.getByUuid(t.getCaze().getUuid())),
```

Comment:
reading the case of each task is not performant, jurisdiction check should be included in the TaskIndexDto  query